client/landing/stepper/declarative-flow/flows/onboarding/onboarding.ts

```diff
@@ -228,72 +165,6 @@ const onboarding: Flow = {
 
 		const submit = async ( providedDependencies: ProvidedDependencies = {} ) => {
 			switch ( currentStepSlug ) {
-				case 'goals': {
-					const goalsUrl = withLocale( '/setup/onboarding/goals', locale );
-					const { intent } = providedDependencies;
-
-					switch ( intent ) {
-						case SiteIntent.Import: {
-							const migrationFlowLink = withLocale( '/setup/hosted-site-migration', locale );
-							return window.location.assign(
-								addQueryArgs( migrationFlowLink, {
-									back_to: goalsUrl,
-								} )
-							);
-						}
```

Comment:
I'm not sure we want to remove all of this. @Automattic/serenity can you validate that?